cpp/arcticdb/version/version_map.hpp

```diff
@@ -802,7 +802,6 @@ class VersionMapImpl {
         auto entry = get_entry(stream_id);
         entry->clear();
         load_via_iteration(store, stream_id, entry, false);
-        fix_stream_ids_of_index_keys(store, stream_id, entry);
```

Comment:
Do we need to keep this `scan_and_rewrite` and `fix_ref_key` functions at all? What are they for?